scripts/mkvirtualenv.bat

```diff
@@ -74,24 +74,31 @@ REM In deactivate.bat, reset PYTHONPATH to its former value
 )
 
 call "%WORKON_HOME%\%ENVNAME%\Scripts\activate.bat"
+if not "%PROJECTDIR%"=="" call setprojectdir.bat "%PROJECTDIR%"
 
 REM Run postmkvirtualenv.bat
 
 if defined VIRTUALENVWRAPPER_HOOK_DIR (
     if exist "%VIRTUALENVWRAPPER_HOOK_DIR%\postmkvirtualenv.bat" (
-    	call "%VIRTUALENVWRAPPER_HOOK_DIR%\postmkvirtualenv.bat"
+        call "%VIRTUALENVWRAPPER_HOOK_DIR%\postmkvirtualenv.bat"
     )
 )
 
 
 goto END
 
-:GET_ENVNAME
-  set "ENVNAME=%~1"
-  shift
-  if not "%~1"=="" goto GET_ENVNAME
+:GET_OPTIONS

```

Comment:
Option parsing in .bat files are a pain, and I'm afraid your solution isn't correct.  The current code passes _all_ parameters to virtualenv.exe, which works because we don't handle any parameters that virtualenv.exe doesn't understand.

What _you_ need is code that removes any `-a xxx`, extracts the name of the new virtualenv, and collects the rest of the parameters so they can be used when calling virtualenv.exe.

I would probably move the getopt code to the top of the file (so you don't need to use `%~1` etc.).

Here is a sketch of the kind of structure I'd be looking for:
```
setlocal enabledelayedexpansion

set PROJECTDIR=
set ENVNAME=
set PASSTRHOUGH=
set FOUNDOPTS=
set VERBOSE=

if "%1"=="" goto print

:getopts
    rem echo GETOPTS "%1, %2, %3, %4, %5, %6, %7, %8, %9"
    set "CUR=%1"
    set OPT=
    if "%CUR:~0,1%"=="-" (
        rem found an option by checking first character equal to -
        set "OPT=%1"
        shift
    )
    if "%OPT%" == "" (
        set "PASSTRHOUGH=%PASSTRHOUGH% %1"
        rem last non-option is envname, keep overwriting the variable
	set "ENVNAME=!CUR!"
        shift
    ) else (
        set FOUNDOPTS=

        if "%OPT%" == "-v" (
            set VERBOSE=1
	    set "PASSTRHOUGH=%PASSTRHOUGH% !CUR!"
            set "FOUNDOPTS=!FOUNDOPTS! %OPT%"
        )

        if "%OPT%" == "-a" (
            set "PROJECTDIR=%1"
            set "FOUNDOPTS=!FOUNDOPTS! %OPT%"
            shift
        )

        rem was it a known option?
        if "!FOUNDOPTS!"=="" (
            set "PASSTRHOUGH=%PASSTRHOUGH% %OPT% %1"
            shift
        )
    )

if not "%1" == "" goto getopts
set FOUNDOPTS=

:print
if not "%VERBOSE%" == "" (
    echo options
    echo --------------------------------
    echo PROJECTDIR="%PROJECTDIR%"
    echo PASSTHROUGH="%PASSTRHOUGH%"
    echo ENVNAME="%ENVNAME%"
)
```